assets/js/googlesitekit/datastore/site/info.test.js

```diff
@@ -74,6 +74,64 @@ describe( 'core/site site info', () => {
 	} );
 
 	describe( 'selectors', () => {
+		describe( 'getAdminURL', () => {
+			it( 'returns the adminURL on its own if no page argument is supplied', async () => {
+				global[ baseInfoVar ] = baseInfo;
+				global[ entityInfoVar ] = entityInfo;
+
+				expect( global[ baseInfoVar ] ).not.toEqual( undefined );
+				expect( global[ entityInfoVar ] ).not.toEqual( undefined );
+
+				let adminURL = registry.select( STORE_NAME ).getAdminURL();
+				expect( adminURL ).toEqual( 'http://something.test/wp-admin' );
+
+				adminURL = registry.select( STORE_NAME ).getAdminURL( undefined, { arg1: 'argument-1', arg2: 'argument-2' } );
+				expect( adminURL ).toEqual( 'http://something.test/wp-admin' );
+			} );
+
+			it( 'returns the adminURL with page query parameter if the page argument is supplied', async () => {
+				global[ baseInfoVar ] = baseInfo;
+				global[ entityInfoVar ] = entityInfo;
+
+				expect( global[ baseInfoVar ] ).not.toEqual( undefined );
+				expect( global[ entityInfoVar ] ).not.toEqual( undefined );
+
+				const adminURL = registry.select( STORE_NAME ).getAdminURL( 'testpage' );
+				expect( adminURL ).toEqual( 'http://something.test/wp-admin?page=testpage' );
```

Comment:
@aaemnnosttv Good catch! I think we should couple this function to `admin.php`, at least by default. We could do the following:
* If you call `getAdminURL( 'googlesitekit-dashboard' )`, you get `{adminURL}/admin.php?page=googlesitekit-dashboard`.
* If you call `getAdminURL( 'admin.php?page=googlesitekit-dashboard' )`, you get the same result.

We would in ~99% of cases use the first since it's a lot simpler, but we would support more custom cases where maybe we want to link to somewhere other than Site Kit pages.

We could use a check for something like `.php?page=` in the `page` parameter to check whether it's the simple format or the "full" format. If someone provides a full path, we should also ensure that it's concatenated with the base admin URL properly, with a single trailing slash.

Reply:
cc @jqlee85

Reply:
@felixarntz just to clarify, it sounds like we're treating the first argument as a special value, which can contain a query string. If the first arg contains a query string, the first part would be used as the path, and the query string would be parsed and merged into the `queryArgs` to be applied (if any). If the first arg did not contain a query string, then the path would default to `admin.php` with the first arg as `queryArgs.page`, which would take precedence over a passed `page` key in the args object. Is that correct?

Reply:
Yep 👍